### ISSUE TITLE
Seg fault when calling backwards on Conv1d

### ISSUE THREAD

**Issue by RPrenger** [pytorch/pytorch#2592]
If you call backwards on a graph that has a `Conv1d` with the `weight` parameter's `requires_grad` flag set to `False` it segfaults with no warning.  This happens even if something downstream still requires gradients.  Here is code that reproduces it (using Pytorch 1.10):
```""" Minimal example showing bug.  Results in segfault"""
import numpy as np
import torch
from torch.autograd import Variable

class Bug(torch.nn.Module):
    def __init__(self):
        super(Bug, self).__init__()

        self.conv = torch.nn.Conv1d(1, 1,
                                    kernel_size=32,
                                    stride = 8,
                                    bias=False)
        if torch.cuda.is_available():
            self.conv.weight.data = self.conv.weight.data.cuda()
        self.conv.weight.requires_grad = False

    def forward(self, signals):
        return self.conv(signals)

if __name__ == '__main__':
    # Create random Tensors for weights, and wrap them in Variables.
    signals = Variable(torch.randn(1, 1, 64), requires_grad=True)

    if torch.cuda.is_available():
        signals.data = signals.data.cuda()

    f=Bug()
    x = f(signals)
    output = torch.sum(f(signals))
    output.backward()

**Comment by albanD** [comment]
Could you try upgrading from your version which is 0.1.10 to 0.2.0 please, I cannot reproduce this with the latest version.
